src/solidity/decodeInfo.js

```diff
@@ -112,10 +113,11 @@ function Array (type, stateDefinitions) {
   * Enum decode the given @arg type
   *
   * @param {String} type - type given by the AST (e.g enum enumDef)
+  * @param {Object} states - all state definitions given by the AST (including struct and enum type declaration) for all contracts
   * @return {Object} returns decoded info about the current type: { storageBytes, typeName, enum}
   */
-function Enum (type, stateDefinitions) {
-  var enumDef = getEnum(type, stateDefinitions)
+function Enum (type, states) {
+  var enumDef = getEnum(type.slice(5), states)
```

Comment:
Please use a regex instead of `slice` - we should check that it starts with `enum `.